pkg/io/serializable.go

```diff
@@ -7,6 +7,14 @@ package io
 // caller should handle the error once and all the other code should just not
 // panic in presence of error.
 type Serializable interface {
+	decodable
+	encodable
```

Comment:
I don't think it's worth the trouble and and in its current form it's gonna look strange from the outside of this module.